demo/src/plugins.rs

```diff
@@ -78,3 +78,40 @@ impl Plugin for CustomShapes {
         );
     }
 }
+
+#[derive(Default, Clone)]
+pub struct ClickWatcher {
+    pub clicked_at: Option<Position>,
+}
+
+impl ClickWatcher {
```

Comment:
Needed to time to figure out how does this stuff work, and honestly, I still don't quite follow it. :)

It handles events i handle_event, then in the next frame, it clones itself, and uses the data from the event to draw the circle?

How about something like that?

```rust
#[derive(Default)]
pub struct ClickWatcher {
    pub clicked_at: Option<Position>,
}

impl ClickWatcher {
    pub fn show_my_position(&self, ui: &egui::Ui) {
        if let Some(clicked_at) = self.clicked_at {
            egui::Window::new("My Position")
                .collapsible(false)
                .resizable(false)
                .title_bar(false)
                .anchor(egui::Align2::CENTER_BOTTOM, [0., -10.])
                .show(ui.ctx(), |ui| {
                    ui.label(format!("{:.04} {:.04}", clicked_at.lon(), clicked_at.lat()))
                        .on_hover_text("my position");
                });
        }
    }
}

impl Plugin for &mut ClickWatcher {
    fn draw(&mut self, response: &Response, painter: Painter, projector: &Projector) {
        if !response.changed() && response.clicked_by(egui::PointerButton::Primary) {
            if let Some(offset) = response.hover_pos().map(|p| p - response.rect.center()) {
                self.clicked_at = Some(projector.unproject(offset));
            }
        }

        if let Some(position) = self.clicked_at {
            painter.circle_filled(projector.project(position).to_pos2(), 5.0, Color32::BLUE);
        }
    }
}
```

Whole logic is now in a single frame. `Plugin::draw` needs to be changed to take `&mut self`, but it's all right I guess.

Reply:
Not agree: then the meaning of the function name is violated.

But we can add something like 

```
trait Plugin {
   ...
   fn handle(&mut self, response: &Response, projector: &Projector) {}
}
````
to use optional trait's member

Reply:
Yeah, fair point. Then maybe it's just `run`?

Reply:
I've updated this branch, can you check?

Reply:
code is checked, works fine

Reply:
if ok, I want a suggestion for better naming

Reply:
I still think it would be better to have a single function - `run`.